### PR TITLE
Eye icon on password fields

### PR DESCRIPTION
**Pull request by KvnMrs**
Issue: https://github.com/blockframes/blockframes/issues/6647

Result (login page):

https://user-images.githubusercontent.com/86494393/168822605-ca9bea87-13d5-4ec0-9490-8fdc1c95bdde.mp4

### CODE REVIEW

File: apps/festival/festival/src/app/event/login/login.component.html [hunk 1]
```diff
@@ -20,7 +20,8 @@ <h1>Welcome to Archipel Market!</h1>
     </mat-form-field>
     <mat-form-field appearance="outline">
       <mat-label>Password</mat-label>
-      <input test-id="password" matInput type="password" placeholder="Password" formControlName="password" required />
+      <input test-id="password" matInput type="password" placeholder="Password" formControlName="password" required [type]="hidePassword ? 'password' : 'text'"/>
+      <mat-icon matSuffix svgIcon="visibility" (click)=passwordVisibility() matTooltip="{{ hidePassword ? 'Show Password' : 'Hide Password' }}"></mat-icon>
```
Comment (RemcoSimonides) [inline apps/festival/festival/src/app/event/login/login.component.html:24]: Did you try this?
"For dev: try to wrap the icon into a mat-icon-button and put the click event on that. That would be better than adding the click event on the icon. If the button doesn't ruin the UI too much."

Also, the icon should change to visibility off

File: apps/festival/festival/src/app/event/login/login.component.ts [hunk 2]
```diff
@@ -19,6 +19,7 @@ export class EventLoginComponent implements OnInit, OnDestroy {
   });
   public buttonText = 'Log in';
   public signinIn = false;
+  public hidePassword: boolean = true;
```
Comment (RemcoSimonides) [inline apps/festival/festival/src/app/event/login/login.component.ts:22]:
```suggestion
  public hidePassword = true;
```
Typescript is smart enough to recognise its type already

File: apps/festival/festival/src/app/event/login/login.component.ts [hunk 3]
```diff
@@ -81,4 +82,8 @@ export class EventLoginComponent implements OnInit, OnDestroy {
     this.service.updateAnonymousCredentials({ role: undefined });
     this.router.navigate(['../../'], { relativeTo: this.route, queryParams: this.route.snapshot.queryParams });
   }
+
+  passwordVisibility() {
+    return this.hidePassword = !this.hidePassword
```
Comment (RemcoSimonides) [inline apps/festival/festival/src/app/event/login/login.component.ts:87]: Could be in the html already as it's short enough
```
<button (click)="hidePassword = !hidePassword"></button>
```

Comment (RemcoSimonides) [reply apps/festival/festival/src/app/event/login/login.component.ts:87]: Apply above comments everywhere ^^

**Comment by RemcoSimonides** [comment]
Funny. I don't think this icon has to change color at all ^^ even if the input field is invalid

File: apps/festival/festival/src/app/event/login/login.component.html (hunk 1 above)
Comment (KvnMrs) [reply apps/festival/festival/src/app/event/login/login.component.html:24]: The CSS look like still ok with mat-icon-button, but when user click 'enter' it clicks on eye now, and not on 'Log in' for exemple

File: apps/festival/festival/src/app/event/login/login.component.html [hunk 4]
```diff
@@ -21,7 +21,10 @@ <h1>Welcome to Archipel Market!</h1>
     <mat-form-field appearance="outline">
       <mat-label>Password</mat-label>
       <input test-id="password" matInput type="password" placeholder="Password" formControlName="password" required [type]="hidePassword ? 'password' : 'text'"/>
-      <mat-icon matSuffix svgIcon="visibility" (click)=passwordVisibility() matTooltip="{{ hidePassword ? 'Show Password' : 'Hide Password' }}"></mat-icon>
+      <button matSuffix mat-icon-button (click)="hidePassword = !hidePassword" [matTooltip]="hidePassword ? 'Show Password' : 'Hide Password'">
+        <mat-icon *ngIf="!hidePassword" svgIcon="visibility_off"></mat-icon>
+        <mat-icon *ngIf="hidePassword" svgIcon="visibility"></mat-icon>
```
Comment (RemcoSimonides) [inline apps/festival/festival/src/app/event/login/login.component.html:26]:
```suggestion
        <mat-icon [svgIcon]="hidePassword ? 'visibility' : 'visibility_off'"></mat-icon>
```

File: libs/ui/src/lib/form/password-confirm/password-confirm.component.html [hunk 5]
```diff
@@ -7,10 +7,13 @@
...
       placeholder="New Password"
       formControlName="password"
-      [type]="hideNewPassword ? 'password' : 'text'"
+      [type]="passwordState.new ? 'password' : 'text'"
       required
     />
-    <mat-icon matSuffix svgIcon="visibility" (click)=newPasswordVisibility() matTooltip="{{ hideNewPassword ? 'Show Password' : 'Hide Password' }}"></mat-icon>
+    <button matSuffix mat-icon-button (click)="passwordState.new = !passwordState.new" [matTooltip]="passwordState.new ? 'Show Password' : 'Hide Password'">
+      <mat-icon *ngIf="!passwordState.new" svgIcon="visibility_off"></mat-icon>
+      <mat-icon *ngIf="passwordState.new" svgIcon="visibility"></mat-icon>
```
Comment (RemcoSimonides) [inline libs/ui/src/lib/form/password-confirm/password-confirm.component.html:15]:
```suggestion
        <mat-icon [svgIcon]="hideNewPassword ? 'visibility' : 'visibility_off'"></mat-icon>
```

File: libs/ui/src/lib/form/password-confirm/password-confirm.component.ts [hunk 6]
```diff
@@ -22,10 +22,10 @@ export class PasswordConfirmComponent implements OnInit, OnDestroy, ControlValue
...
 
   public passwordsMatcher: RepeatPasswordStateMatcher;
-
-  public hideNewPassword: boolean = true;
-  public hideConfirmationPassword: boolean = true;
-
+  public passwordState = {
+    new: true,
+    confirm: true
+  }
```
Comment (RemcoSimonides) [inline libs/ui/src/lib/form/password-confirm/password-confirm.component.ts:28]: Let's keep it simple.
```
hideConfirmPassword = true
hideNewPassword = true
```

The name "passwordState" confuses me. It doesn't indicate that it's about hide/show functionality and "passwordState.new" sounds like password field is in a new state or something? so yea, keep it simple ^^

File: libs/user/src/lib/auth/components/signin-form/signin-form.component.html [hunk 7]
```diff
@@ -15,7 +15,10 @@ <h1>Log in</h1>
       <mat-form-field appearance="outline">
         <mat-label>Password</mat-label>
         <input matInput type="password" placeholder="Password" formControlName="password" required test-id="password" [type]="hidePassword ? 'password' : 'text'"/>
-        <mat-icon matSuffix svgIcon="visibility" (click)=passwordVisibility() matTooltip="{{ hidePassword ?  'Show Password' : 'Hide Password' }}"></mat-icon>
+        <button matSuffix mat-icon-button (click)="hidePassword = !hidePassword" [matTooltip]="hidePassword ? 'Show Password' : 'Hide Password'"> 
+          <mat-icon *ngIf="!hidePassword" svgIcon="visibility_off"></mat-icon>
+          <mat-icon *ngIf="hidePassword" svgIcon="visibility"></mat-icon>
```
Comment (RemcoSimonides) [inline libs/user/src/lib/auth/components/signin-form/signin-form.component.html:20]:
```suggestion
        <mat-icon [svgIcon]="hidePassword ? 'visibility' : 'visibility_off'"></mat-icon>
```

File: libs/ui/src/lib/form/password-confirm/password-confirm.component.html [hunk 8]
```diff
@@ -31,10 +34,13 @@
...
       formControlName="confirm"
       [errorStateMatcher]="passwordsMatcher"
-      [type]="hideConfirmationPassword ? 'password' : 'text'"
+      [type]="passwordState.confirm ? 'password' : 'text'"
       required
     />
-    <mat-icon matSuffix svgIcon="visibility" (click)=confirmationPasswordVisibility() matTooltip="{{ hideConfirmationPassword ? 'Show Password' : 'Hide Password' }}"></mat-icon>
+    <button matSuffix mat-icon-button (click)="passwordState.confirm = !passwordState.confirm" [matTooltip]="passwordState.confirm ? 'Show Password' : 'Hide Password'">
+      <mat-icon *ngIf="!passwordState.confirm" svgIcon="visibility_off"></mat-icon>
+      <mat-icon *ngIf="passwordState.confirm" svgIcon="visibility"></mat-icon>
```
Comment (RemcoSimonides) [inline libs/ui/src/lib/form/password-confirm/password-confirm.component.html:42]:
```suggestion
        <mat-icon [svgIcon]="hideConfirmPassword ? 'visibility' : 'visibility_off'"></mat-icon>
```

File: libs/user/src/lib/auth/forms/password/password.component.ts [hunk 9]
```diff
@@ -16,20 +16,10 @@ export class PasswordFormComponent {
...
-    return this.hideCurrentPassword = !this.hideCurrentPassword
-  }
-
-  newPasswordVisibility() {
-    return this.hideNewPassword = !this.hideNewPassword
-  }
-
-  confirmationPasswordVisibility() {
-    return this.hideConfirmationPassword = !this.hideConfirmationPassword
+  public passwordState = {
```
Comment (RemcoSimonides) [inline libs/user/src/lib/auth/forms/password/password.component.ts:19]: Same as above. Keep it simple.

**Comment by KvnMrs** [comment]
@RemcoSimonides
So when the password field is invalid, the eye automatically turns red and gets a margin-bottom of 4px, which offsets it.
This behaviour is due to the file `styles\components\mat-form-field.scss` and the code Line 34 to 41 :
```
.mat-icon {
        svg {
          margin-bottom: 4px;
          path {
            fill: mat.get-color-from-palette($warn);
          }
        }
      }
```
I tried lot of solutions, but i'm still blocked to override this behaviour, the only solution i found is to modify it in the file directly, but some other icons on the app are affected too.

https://user-images.githubusercontent.com/86494393/169023727-25a98f51-66f6-4f06-8389-26887232ca5f.mp4

**Comment by RemcoSimonides** [comment]
I can think of three solutions for this issue:
1. Add a class to the icon and then to add :not() in the builder like
```
.mat-icon:not(.class) {
  svg {
    margin-bottom: 4px;
    path {
      fill: mat.get-color-from-palette($warn);
    }
  }
}
```
But this is not good practice... Could we make an exception Francois?
2. Ask design team whether we could put the icon behind the password fields instead of within.
3. Hope @GrandSchtroumpf has more ideas :)

**Comment by RemcoSimonides** [comment]
Is the PR ready for review?

**Comment by KvnMrs** [comment]
> Is the PR ready for review? or are you working on it?

Yes, still working on it, it stay a last step: show every password fields on page and visibly a correcting for the test

**Comment by RemcoSimonides** [comment]
> > Is the PR ready for review? or are you working on it?
> 
> Yes, still working on it, it stay a last step: show every password fields on page (when there are several) and a correcting for the test

I don't think that showing every password field is a good idea anymore as you changed this to a component. Before it was a good idea because it made the code even easier, now it will only make the code more complicated...

**Comment by KvnMrs** [comment]
@RemcoSimonides 

Now :

https://user-images.githubusercontent.com/86494393/169806467-d7351494-0522-4b84-af4b-91a90ef8455e.mp4

Now :

there is still an unexpected behaviour with the eye and the passwordmatcher.

If this PR is ok, as seen with @GrandSchtroumpf I will create an issue by tagging Anastasiia and Daniela to discuss the following problems with a dev: 

- Review the expected behaviour of the eye (color) ?
- Inform them that a margin botom intended to align icons with text in invalid fields has been removed. Is it essential ? if yes, put it back differently

concerned file: styles\components\mat-form-field.scss

**Comment by RemcoSimonides** [comment]
> > I don't think that showing every password field is a good idea anymore as you changed this to a component. Before it was a good idea because it made the code even easier, now it will only make the code more complicated...
> 
> Sorry, i didnt see your reply before :/. i just found a solution. Can i push it after some quick clean? And let see if its ok or not ? if it's not i will remove it

Well, the code isn't the problem. The problem is that you're adding code for something that isn't a problem. Sorry for the confusion but this feature was really only there because of code, not because of a request from Business/UX/UI 😅 So better remove it again

**Comment by KvnMrs** [comment]
> Well, the code isn't the problem. The problem is that you're adding code for something that isn't a problem. Sorry for the confusion but this feature was really only there because of code, not because of a request from Business/UX/UI 😅 So better remove it again

ok, no problem, i gonna remove it

File: apps/festival/festival/src/app/event/login/login.component.ts [hunk 10]
```diff
@@ -81,4 +81,5 @@ export class EventLoginComponent implements OnInit, OnDestroy {
     this.service.updateAnonymousCredentials({ role: undefined });
     this.router.navigate(['../../'], { relativeTo: this.route, queryParams: this.route.snapshot.queryParams });
   }
+
```
Comment (RemcoSimonides) [inline apps/festival/festival/src/app/event/login/login.component.ts:84]:
```suggestion
```

File: libs/ui/src/lib/form/password-confirm/password-confirm.component.ts [hunk 11]
```diff
@@ -52,4 +52,5 @@ export class PasswordConfirmComponent implements OnInit, OnDestroy, ControlValue
   public registerOnTouched(fn: (touched: boolean) => void) {
     fn(true);
   }
+
```
Comment (RemcoSimonides) [inline libs/ui/src/lib/form/password-confirm/password-confirm.component.ts:55]:
```suggestion
```

File: libs/ui/src/lib/form/password-confirm/password-confirm.module.ts [hunk 12]
```diff
@@ -11,6 +11,10 @@ import { MatChipsModule } from '@angular/material/chips';
 import { MatAutocompleteModule } from '@angular/material/autocomplete';
 import { MatIconModule } from '@angular/material/icon';
 import { MatInputModule } from '@angular/material/input';
+import { MatTooltipModule } from '@angular/material/tooltip';
+
+// @Blockframes
```
Comment (RemcoSimonides) [inline libs/ui/src/lib/form/password-confirm/password-confirm.module.ts:16]:
```suggestion
// Blockframes
```

File: libs/ui/src/lib/form/password-input/password-input.component.ts [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+  styleUrls: ['./password-input.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+
+export class PasswordInputComponent {
+  @Input() public placeholder: string;
+  @Input() public control = new FormControl();
+  @Input() public errorStateMatcher: ErrorStateMatcher;
+  @Input() public label: string;
+  @Input() public hidePassword = true
```
Comment (RemcoSimonides) [inline libs/ui/src/lib/form/password-input/password-input.component.ts:18]:
```suggestion
  public hidePassword = true;
```

File: libs/ui/src/lib/form/password-input/password-input.component.ts [hunk 14]
```diff
@@ -0,0 +1,20 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { ErrorStateMatcher } from '@angular/material/core';
+import { boolean } from '@blockframes/utils/decorators/decorators';
+
+@Component({
+  selector: 'password-input',
```
Comment (RemcoSimonides) [inline libs/ui/src/lib/form/password-input/password-input.component.ts:7]:
```suggestion
  selector: '[control] password-input',
```
Makes the control input mandatory

File: libs/ui/src/lib/form/password-input/password-input.component.ts [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+  templateUrl: './password-input.component.html',
+  styleUrls: ['./password-input.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+
```
Comment (RemcoSimonides) [inline libs/ui/src/lib/form/password-input/password-input.component.ts:12]:
```suggestion
```

File: libs/ui/src/lib/form/password-input/password-input.component.ts [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+export class PasswordInputComponent {
+  @Input() public placeholder: string;
+  @Input() public control = new FormControl();
```
Comment (RemcoSimonides) [inline libs/ui/src/lib/form/password-input/password-input.component.ts:15]: Don't initialise the value as it has to be passed in through the Input

File: libs/ui/src/lib/form/password-input/password-input.module.ts [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatIconModule } from '@angular/material/icon';
+import { MatTooltipModule } from '@angular/material/tooltip';
+
+@NgModule({
+  declarations: [PasswordInputComponent],
+  exports: [PasswordInputComponent],
+  imports: [
+    CommonModule,
+    FlexLayoutModule,
```
Comment (RemcoSimonides) [inline libs/ui/src/lib/form/password-input/password-input.module.ts:20]: Not needed

File: libs/user/src/lib/auth/components/signin-form/signin-form.component.ts [hunk 18]
```diff
@@ -9,7 +9,9 @@ import { SigninForm } from '../../forms/signin.form';
 })
 
 export class SigninFormComponent {
-  @Output() submited = new EventEmitter();
 
+  @Output() submited = new EventEmitter();
```
Comment (RemcoSimonides) [inline libs/user/src/lib/auth/components/signin-form/signin-form.component.ts:13]: Undo changes on this page.

File: libs/user/src/lib/auth/components/signin-form/signin-form.module.ts [hunk 19]
```diff
@@ -10,10 +10,14 @@ import { MatButtonModule } from '@angular/material/button';
 import { MatFormFieldModule } from '@angular/material/form-field';
 import { MatInputModule } from '@angular/material/input';
 import { MatCardModule } from '@angular/material/card';
+import { MatIconModule } from '@angular/material/icon';
```
Comment (RemcoSimonides) [inline libs/user/src/lib/auth/components/signin-form/signin-form.module.ts:13]: Remove imports which aren't used

File: libs/user/src/lib/auth/forms/password/password.component.html [hunk 20]
```diff
@@ -4,27 +4,24 @@
...
-        <input
-          test-id="password"
-          matInput
-          type="password"
-          placeholder="New Password"
-          formControlName="next"
-          [errorStateMatcher]="currentPasswordMatch"
-          required
-        />
+      <!-- Current Password field -->
```
Comment (RemcoSimonides) [inline libs/user/src/lib/auth/forms/password/password.component.html:7]: As it says 3x Current Password in the code below, I think this comment is redundant

File: libs/user/src/lib/auth/forms/password/password.component.html [hunk 21]
```diff
@@ -4,27 +4,24 @@
...
+      <password-input
+        [control]="form.get('current')"
+        placeholder="Current Password"
+        label="Current Password"
+        test-id="current-password"
+        hidePassword="hidePassword"
+      >
+      </password-input>
+      <!-- New Password field -->
```
Comment (RemcoSimonides) [inline libs/user/src/lib/auth/forms/password/password.component.html:16]: As it says 2x New Password in the code below, I think this comment is redundant

File: libs/user/src/lib/auth/forms/password/password.component.html [hunk 22]
```diff
@@ -38,26 +35,24 @@
...
-          test-id="password-confirm"
-          matInput
-          type="password"
-          placeholder="New Password Confirmation"
-          formControlName="confirm"
-          [errorStateMatcher]="passwordsMatcher"
-          required
-        />
+      </password-input>
+      <!-- Confirmation New Password field -->
```
Comment (RemcoSimonides) [inline libs/user/src/lib/auth/forms/password/password.component.html:39]: As it says 2x New Password Confirmation in the code below, I think this comment is redundant

File: libs/user/src/lib/auth/forms/password/password.module.ts [hunk 23]
```diff
@@ -4,22 +4,30 @@ import { FlexLayoutModule } from '@angular/flex-layout';
 import { ReactiveFormsModule } from '@angular/forms';
 
 import { PasswordFormComponent } from './password.component';
+import { PasswordInputModule } from '@blockframes/ui/form/password-input/password-input.module';
 
 // Material
 import { MatCardModule } from '@angular/material/card';
 import { MatFormFieldModule } from '@angular/material/form-field';
 import { MatInputModule } from '@angular/material/input';
+import { MatIconModule } from '@angular/material/icon';
```
Comment (RemcoSimonides) [inline libs/user/src/lib/auth/forms/password/password.module.ts:13]: Remove unused imports

File: libs/user/src/lib/auth/pages/identity/identity.component.html [hunk 24]
```diff
@@ -95,17 +95,14 @@ <h1>Create your account</h1>
...
-          <input
-            matInput
-            type="password"
-            placeholder="New Password"
-            formControlName="password"
-            [errorStateMatcher]="currentPasswordMatch"
-            required
-            test-id="password"
-          />
+        <!-- NEW PASSWORD -->
```
Comment (RemcoSimonides) [inline libs/user/src/lib/auth/pages/identity/identity.component.html:98]: Reduntant comment

File: libs/user/src/lib/auth/pages/identity/identity.component.html [hunk 25]
```diff
@@ -116,29 +113,25 @@ <h1>Create your account</h1>
...
-            type="password"
-            placeholder="New Password Confirmation"
-            formControlName="confirm"
-            [errorStateMatcher]="passwordsMatcher"
-            required
-            test-id="password-confirm"
-          />
+        </password-input>
+
+        <!-- PASSWORD CONFIRMATION -->
```
Comment (RemcoSimonides) [inline libs/user/src/lib/auth/pages/identity/identity.component.html:118]: Redundant comment

File: libs/utils/src/lib/form/matchers.ts [hunk 26]
```diff
@@ -10,16 +10,14 @@ export class CrossFieldErrorMatcher implements ErrorStateMatcher {
...
-  private password: string;
-  private confirm: string;
-
-  constructor(password: string = 'password', confirm: string = 'confirm') {
-    this.password = password;
-    this.confirm = confirm;
-  }
+  constructor(
+    private password: string = 'password',
+    private confirm: string = 'confirm'
```
Comment (RemcoSimonides) [inline libs/utils/src/lib/form/matchers.ts:15]:
```suggestion
    private password = 'password',
    private confirm = 'confirm'
```

Comment (RemcoSimonides) [reply libs/utils/src/lib/form/matchers.ts:15]: Nice refactor

File: libs/user/src/lib/auth/forms/password/password.component.html [hunk 27]
```diff
@@ -4,27 +4,24 @@
...
-          [errorStateMatcher]="currentPasswordMatch"
-          required
-        />
+      <!-- Current Password field -->
+      <password-input
+        [control]="form.get('current')"
+        placeholder="Current Password"
+        label="Current Password"
+        test-id="current-password"
+        hidePassword="hidePassword"
```
Comment (RemcoSimonides) [inline libs/user/src/lib/auth/forms/password/password.component.html:13]: remove hidePassword input

File: libs/user/src/lib/auth/forms/password/password.component.html [hunk 28]
```diff
@@ -4,27 +4,24 @@
...
+      >
+      </password-input>
+      <!-- New Password field -->
+      <password-input 
+        [control]="form.get('next')"
+        [errorStateMatcher]="currentPasswordMatch"
+        placeholder="New Password"
+        label="New Password"
+        test-id="password"
+        hidePassword="hidePassword"
```
Comment (RemcoSimonides) [inline libs/user/src/lib/auth/forms/password/password.component.html:23]: remove hidePassword input

File: libs/user/src/lib/auth/forms/password/password.component.html [hunk 29]
```diff
@@ -38,26 +35,24 @@
...
-        />
+      </password-input>
+      <!-- Confirmation New Password field -->
+      <password-input
+        [control]="form.get('confirm')"
+        [errorStateMatcher]="passwordsMatcher"
+        placeholder="New Password Confirmation"
+        label="New Password Confirmation"
+        test-id="password-confirm"
+        hidePassword="hidePassword"
```
Comment (RemcoSimonides) [inline libs/user/src/lib/auth/forms/password/password.component.html:46]: remove hidePassword input

File: libs/user/src/lib/auth/pages/identity/identity.component.html [hunk 30]
```diff
@@ -95,17 +95,14 @@ <h1>Create your account</h1>
...
-            required
-            test-id="password"
-          />
+        <!-- NEW PASSWORD -->
+        <password-input
+          [control]="form.get('password')"
+          placeholder="New Password"
+          label="Password"
+          test-id="password"
+          hidePassword="hidePassword"
```
Comment (RemcoSimonides) [inline libs/user/src/lib/auth/pages/identity/identity.component.html:104]: remove hidePassword input

File: libs/user/src/lib/auth/pages/identity/identity.component.html [hunk 31]
```diff
@@ -116,29 +113,25 @@ <h1>Create your account</h1>
...
+        </password-input>
+
+        <!-- PASSWORD CONFIRMATION -->
+        <password-input
+          [control]="form.get('confirm')"
+          [errorStateMatcher]="passwordsMatcher"
+          placeholder="New Password Confirmation"
+          label="Password Confirmation"
+          test-id="password-confirm"
+          hidePassword="hidePassword"
```
Comment (RemcoSimonides) [inline libs/user/src/lib/auth/pages/identity/identity.component.html:125]: remove hidePassword input

File: libs/ui/src/lib/form/password-input/password-input.component.ts [hunk 32]
```diff
@@ -0,0 +1,19 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { ErrorStateMatcher } from '@angular/material/core';
+import { boolean } from '@blockframes/utils/decorators/decorators';
```
Comment (phpgeekfr) [inline libs/ui/src/lib/form/password-input/password-input.component.ts:4]: remove unsed imports

File: libs/ui/src/lib/form/password-input/password-input.component.html [hunk 33]
```diff
@@ -0,0 +1,21 @@
+<mat-form-field appearance="outline">
+  <mat-label>{{ label }}</mat-label>
+  <input 
+    matInput
+    [placeholder]="placeholder"
+    [formControl]="control"
+    [errorStateMatcher]="errorStateMatcher"
+    required
+    [type]="hidePassword ? 'password' : 'text'"
+    test-id="testId"
```
Comment (RemcoSimonides) [inline libs/ui/src/lib/form/password-input/password-input.component.html:10]: This doesn't work. Without using `[]` you're just passing a string to it.
```
[test-id]="testId"
```

File: libs/utils/src/lib/form/matchers.ts [hunk 34]
```diff
@@ -10,44 +10,26 @@ export class CrossFieldErrorMatcher implements ErrorStateMatcher {
...
-  private password: string;
-  private confirm: string;
-
-  constructor(password: string = 'password', confirm: string = 'confirm') {
-    this.password = password;
-    this.confirm = confirm;
-  }
+  constructor(
+    private password = 'password',
+    private confirm = 'confirm'
```
Comment (phpgeekfr) [inline libs/utils/src/lib/form/matchers.ts:15]: missing `;`

File: libs/user/src/lib/auth/pages/profile/profile.module.ts [hunk 35]
```diff
@@ -5,7 +5,7 @@ import { FlexLayoutModule } from '@angular/flex-layout';
 
 // Modules
 import { ProfileFormModule } from '@blockframes/auth/forms/profile/profile.module';
-import { PasswordFormModule } from '@blockframes/auth/forms/password/password.module';
+import { PasswordFormModule } from '@blockframes/auth/forms/password/change-password.module';
```
Comment (phpgeekfr) [inline libs/user/src/lib/auth/pages/profile/profile.module.ts:8]: also rename the module accordingly

File: libs/user/src/lib/auth/pages/identity/identity.component.html [hunk 36]
```diff
@@ -95,17 +95,12 @@ <h1>Create your account</h1>
           </div>
```
Comment (phpgeekfr) [inline libs/user/src/lib/auth/pages/identity/identity.component.html:95]: use `<password-input` for `invitation pass`

File: libs/user/src/lib/auth/forms/password/change-password.module.ts [hunk 37]
```diff
@@ -3,23 +3,23 @@ import { CommonModule } from '@angular/common';
...
 @NgModule({
   declarations: [PasswordFormComponent],
   imports: [
     CommonModule,
     FlexLayoutModule,
     ReactiveFormsModule,
+    PasswordInputModule,
     // Material
     MatCardModule,
     MatFormFieldModule,
```
Comment (phpgeekfr) [inline libs/user/src/lib/auth/forms/password/change-password.module.ts:22]: I think `MatFormFieldModule,` can be removed

File: libs/user/src/lib/auth/pages/identity/identity.component.html (hunk 36 above)
Comment (KvnMrs) [reply libs/user/src/lib/auth/pages/identity/identity.component.html:95]: Ok, but I also left a comment about this issue by tagging Anastasiia this morning

File: libs/user/src/lib/auth/forms/password/change-password.module.ts (hunk 37 above)
Comment (KvnMrs) [reply libs/user/src/lib/auth/forms/password/change-password.module.ts]: It's needed for the `<mat-hint>`

File: libs/utils/src/lib/form/matchers.ts (hunk 26 above)
[the remark this answers is not in the document]
Comment (KvnMrs) [reply libs/utils/src/lib/form/matchers.ts]: I can't now with the new refact, the ' , ' is needed

**Pull request merged**